crates/sncast/src/starknet_commands/script.rs

```diff
@@ -375,49 +256,53 @@ impl CairoHintProcessor<'_> {
                     },
                 ))?;
 
-                buffer
-                    .write(Felt252::from_(invoke_response.transaction_hash.0))
-                    .expect("Failed to insert transaction hash");
-
-                Ok(())
+                let res: Vec<Felt252> = vec![Felt252::from_(invoke_response.transaction_hash.0)];
+                Ok(CheatcodeHandlingResult::Handled(res))
             }
             "get_nonce" => {
                 let block_id = as_cairo_short_string(&inputs[0])
                     .expect("Failed to convert entry point name to short string");
-                let account = self.runtime.block_on(get_account(
+                let account = self.tokio_runtime.block_on(get_account(
                     &self.config.account,
                     &self.config.accounts_file,
                     self.provider,
                     self.config.keystore.clone(),
                 ))?;
 
-                let nonce = self.runtime.block_on(get_nonce(
+                let nonce = self.tokio_runtime.block_on(get_nonce(
                     self.provider,
                     &block_id,
                     account.address(),
                 ))?;
-                buffer
-                    .write(Felt252::from_(nonce))
-                    .expect("Failed to insert nonce");
 
-                Ok(())
+                let res: Vec<Felt252> = vec![Felt252::from_(nonce)];
+                Ok(CheatcodeHandlingResult::Handled(res))
             }
-            _ => Err(anyhow!("Unknown cheatcode selector: {selector}")),
+            _ => Err(EnhancedHintError::from(anyhow!(
+                "Cheatcode {selector} not implemented"
```

Comment:
I'm not sure we should use `cheatcode` naming in errors and messages to users in cast tbh. I find it a bit misleading, as this is something that immidiately makes me think of forge - in cast we have subcommands (or just functions in cast_std) instead. Perhaps we should stick to this naming? I'm not 100% sure, so perhaps let's discuss.

Reply:
I think we can stick to using `subcommands` in deployment scripts.

Reply:
Other options I'm thinking of are:
- "Selector {selector} not implemented"
- "{selector} not implemented"

Reply:
But we can use `subcommands` for now

Reply:
this error originates in scripts, so I think `functions` may have more sense. this is just a nitpick though, if you like subcommands more I won't oppose.

Reply:
I think cheatcode is the name of the cairo function, nothing cast/forge related, you shouldn't handle it at all here, some extensions under this one might be able to handle it.